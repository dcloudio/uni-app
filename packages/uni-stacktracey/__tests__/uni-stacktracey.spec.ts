const path = require('path')
const {
  stacktracey,
  uniStracktraceyPreset,
  utsStracktraceyPreset,
} = require('../dist/uni-stacktracey.cjs.js')

const utsProjectDir = path.resolve(__dirname, '../test/uts')

const utsErrorMsg = `Error: 
${path.resolve(
  utsProjectDir,
  'unpackage/dist/dev/app-plus/uni_modules/test-uts/app-android/index.kt'
)}:59:67: error: unresolved reference: UtsJSONObject
open suspend fun testClassAsync(opts: AsyncOptions): Deferred<UtsJSONObject> = CoroutineScope(Dispatchers.Default).async {
                                                              ^
`

const uniErrorMsg = `Error: Sentry Error
at a.throwError(/static/js/pages-index-index.3ab0d0e5.js:1:567)
at click(/static/js/pages-index-index.3ab0d0e5.js:1:2745)
at ee(/static/js/chunk-vendors.75525bd5.js:34:11927)
at n(/static/js/chunk-vendors.75525bd5.js:34:13747)
at ee(/static/js/chunk-vendors.75525bd5.js:34:11927)
at HTMLElement.n(/static/js/chunk-vendors.75525bd5.js:34:13824)
at HTMLElement.o._wrapper(/static/js/chunk-vendors.75525bd5.js:34:53966)
at HTMLElement.i(/static/js/chunk-vendors.75525bd5.js:7:609894)`

describe('uni-stacktracey', () => {
  test('uniStracktraceyPreset local', () => {
    stacktracey(uniErrorMsg, {
      preset: uniStracktraceyPreset({
        base: path.resolve(__dirname, '../test/__UNI_APPID__/h5/1.0.0/'),
        sourceRoot: '',
      }),
    }).then((res: string) => {
      expect(res).toEqual(`Error: Sentry Error
at   src/pages/index/index.vue:44
at   src/pages/index/index.vue?be58:12:17
at   node_modules/@sentry/browser/esm/helpers.js:74:22
at   node_modules/@dcloudio/vue-cli-plugin-uni/packages/h5-vue/dist/vue.runtime.esm.js:1864:25
at   node_modules/@dcloudio/vue-cli-plugin-uni/packages/h5-vue/dist/vue.runtime.esm.js:2189:13
at   node_modules/@dcloudio/vue-cli-plugin-uni/packages/h5-vue/dist/vue.runtime.esm.js:1864:25
at   node_modules/@dcloudio/vue-cli-plugin-uni/packages/h5-vue/dist/vue.runtime.esm.js:2185:8
at   node_modules/@dcloudio/vue-cli-plugin-uni/packages/h5-vue/dist/vue.runtime.esm.js:7076:24`)
    })
  })

  test('utsStracktraceyPreset local', () => {
    stacktracey(utsErrorMsg, {
      preset: utsStracktraceyPreset({
        inputRoot: '/Users/fxy/Projects/Demo/my-vue3-project-uts/src',
        outputRoot: path.resolve(utsProjectDir, 'unpackage/dist/dev/app-plus'),
        sourceMapRoot: path.resolve(
          utsProjectDir,
          'unpackage/dist/dev/.sourcemap/app-plus'
        ),
      }),
    }).then((res: string) => {
      expect(res).toEqual(`Error:
at uni_modules/test-uts/app-android/index.uts:82:52
error: unresolved reference: UtsJSONObject
open suspend fun testClassAsync(opts: AsyncOptions): Deferred<UtsJSONObject> = CoroutineScope(Dispatchers.Default).async {
^
`)
    })
  })
})
