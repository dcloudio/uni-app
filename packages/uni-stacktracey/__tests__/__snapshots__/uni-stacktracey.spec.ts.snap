// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uni-stacktracey parse arkts stacktrace 1`] = `
"at uni_modules/native-button/utssdk/app-harmony/index.uts:91:47

Argument of type 'UniNativeViewElement | null' is not assignable to parameter of type 'UniNativeViewElement'.
Type 'null' is not assignable to type 'UniNativeViewElement'."
`;

exports[`uni-stacktracey parse arkts stacktrace 2`] = `
"at uni_modules/native-button/utssdk/app-harmony/builder.ets:8:9

Cannot find name 'param'. Did you mean 'params'?"
`;

exports[`uni-stacktracey parse arkts stacktrace 3`] = `
"at D:/demo-projects/demo-app-x/unpackage/debug/app-harmony-fde54b25/oh_modules/.ohpm/@dcloudio+uni-app-x-runtime@8kwvevyo0ovlx9ebtjtkdkzydpkx+j5qpbuuwz5vjge=/oh_modules/@dcloudio/uni-app-x-runtime/src/main/ets/runtime/dom/NamedNodeMap.ets:16:45
"
`;

exports[`uni-stacktracey parse kotlin stacktrace 1`] = `
"Error:
at uni_modules/test-uts/app-android/index.uts:82:52
error: unresolved reference: UTSJSONObject
open suspend fun testClassAsync(opts: AsyncOptions): Deferred<UTSJSONObject> = CoroutineScope(Dispatchers.Default).async {
^
"
`;

exports[`uni-stacktracey parse swift stacktrace 1`] = `
"at uni_modules/test-uts1/utssdk/app-ios/index.uts:2:10
error: cannot convert return expression of type 'Int' to return type 'String'
at uni_modules/test-uts1/utssdk/app-ios/index.uts:5:10
error: cannot convert return expression of type 'Int' to return type 'String'
"
`;

exports[`uni-stacktracey parse web stacktrace 1`] = `
"Error: Sentry Error
at   src/pages/index/index.vue:44:0
at   src/pages/index/index.vue?be58:12:17
at   node_modules/@sentry/browser/esm/helpers.js:74:22
at   node_modules/@dcloudio/vue-cli-plugin-uni/packages/h5-vue/dist/vue.runtime.esm.js:1864:25
at   node_modules/@dcloudio/vue-cli-plugin-uni/packages/h5-vue/dist/vue.runtime.esm.js:2189:13
at   node_modules/@dcloudio/vue-cli-plugin-uni/packages/h5-vue/dist/vue.runtime.esm.js:1864:25
at   node_modules/@dcloudio/vue-cli-plugin-uni/packages/h5-vue/dist/vue.runtime.esm.js:2185:8
at   node_modules/@dcloudio/vue-cli-plugin-uni/packages/h5-vue/dist/vue.runtime.esm.js:7076:24"
`;
